.travis.yml

```diff
@@ -1,12 +1,30 @@
 language: node_js
 node_js:
   - "lts/*"
-cache: yarn
-script:
-  - yarn run lint
-  - yarn test
-  - yarn run coverage
-  - yarn build:docs
+
+cache:
+  yarn: true
+  directories:
+    - node_modules
+
+jobs:
+  include:
+    - stage: Lint and Check types
+      name: Check types
+      script: yarn run check-types & yarn run lint
+
+    - stage: Tests
+      name: Unit
+      script: yarn run test:unit && yarn run codecov
```

Comment:
Not really for this PR, but it might be good to run codecov across all our tests.
I know we have tests in e2e that would expand our coverage. And the spec tests would expand our coverage a lot too.

Reply:
It might be missleading as in e2e tests we test some feature and we might cover code that actually doesn''t work but we didn't test if it works correctly. I think codecov is more useful for unit tests. Anyways, better in separate PR as there is already issue to fix wrong reports